diem-move/diem-framework/experimental/releases/artifacts/current/build/DiemExperimental/sources/Roles.move

```diff
@@ -192,9 +193,12 @@ module DiemFramework::Roles {
     // Role Checking
 
     fun has_role(account: &signer, role_id: u64): bool acquires RoleId {
-       let addr = Signer::address_of(account);
-       exists<RoleId>(addr)
-           && borrow_global<RoleId>(addr).role_id == role_id
+        let addr = Signer::address_of(account);
+        if (!exists<RoleId>(addr)) {
+            role_id == NO_ROLE_ID
+        } else {
+            borrow_global<RoleId>(addr).role_id == role_id
+        }
```

Comment:
This seems overly complicated.  Why isn't it just `role_id == get_role_id(addr)`?